### ISSUE TITLE
use gradle nativeBuild instead of gradle build for java lambda function build with Graal

### ISSUE THREAD

**Issue by marcelweigle** [aws-amplify/amplify-console#2228]
**Please describe which feature you have a question about?**
In amplify cicd in the backend section I need to use gradle nativeImage to build my lambda function. Probably gradle build is executed instead by default. Can I somehow in the amplify.yml specify that gradle nativeImage should be used?

this is my build.gradle file
plugins {
    id 'org.springframework.boot' version '2.5.3'
    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
    id 'java'
    id 'org.springframework.experimental.aot' version '0.10.2'
    id 'org.graalvm.buildtools.native' version '0.9.1'
}

group = 'example'

repositories {
    maven { url 'https://repo.spring.io/release' }
    maven { url 'https://repo.spring.io/milestone' }
    maven { url 'https://repo.spring.io/snapshot' }
    mavenCentral()
}

dependencies {
    implementation('org.springframework.cloud:spring-cloud-function-adapter-aws:3.1.3')
    implementation 'org.springframework.cloud:spring-cloud-starter-function-web:3.1.3'
    implementation('org.springframework.boot:spring-boot-configuration-processor:2.5.3')
    compileOnly("com.amazonaws:aws-lambda-java-events:3.9.0")
    compileOnly("com.amazonaws:aws-lambda-java-core:1.2.1")
    testImplementation('org.springframework.boot:spring-boot-starter-test:2.5.3')
}

test {
    useJUnitPlatform()
}

nativeBuild {
    classpath "$buildDir/resources/aot", "$buildDir/classes/java/aot"
}

task buildZip(type: Zip) {
    from compileJava
    from processResources
    into('lib') {
        from configurations.runtimeClasspath
    }
    archiveFileName = 'latest_build.zip'
    destinationDirectory = file("$buildDir/distributions")

    from ("$buildDir/native/dealstatus", "bootstrap")
}

task buildZipDist(type: Zip) {
    from compileJava
    from processResources
    into('lib') {
        from configurations.runtimeClasspath
    }
    archiveFileName = 'latest_build.zip'
    destinationDirectory = file("/dist")

    from ("$buildDir/native/dealstatus", "bootstrap")
}
build.dependsOn buildZip
build.dependsOn buildZipDist

**Comment by behrooziAWS** [comment]
I'm not sure I follow. Are you saying that the gradle command that you have put in your amplify.yml today is causing gradle to build without nativeImage?  Can you provide appid/region/branch/build id so we can review?

I'm not familiar with Graal or Gradle.  When building with the build.gradle file locally using the same gradle build command does it use nativeImage?

**Comment by marcelweigle** [comment]
appid: angeldesk.io, region EU-Central-1, branch dev (backend function dealStatus)

Yes executing the command "gradle build" - which is the standard to build the project with gradle - the native images does not get build (only normal jar)
Instead the command "gradle nativeBuild" has to be used. 

It is the same locally. You can build a native image locally by using "gradle nativeBuild". The Problem is that I have a Mac and because of that it is a "Mac native image" which cannot be run on AWS Lambda because AWS Lambda runs on Linux. But as I know amplifys cicd  uses aws code build and runs on linux and therefore the native image created is a linux native image and can be run on AWS Lambda.

To sum it up: In the amplify.yml I need to specify that amplify cicd uses "gradle nativeBuild" to build my dealStatus lambda function and not the default "gradle build".

**Comment by Jupdi** [comment]
The problem is caused here: https://github.com/aws-amplify/amplify-cli/blob/f2fa4794d4bcf011ffcf2fa66de7b102c034a437/packages/amplify-java-function-runtime-provider/src/utils/build.ts#L23
The runtime provider uses hard coded gradle build. That should be configurable.
